### PR TITLE
raftstore: skip deleting snapshot files in peer pending_remove is true (#11782)

### LINKED ISSUE #11746: raftstore: snapshot files may deleted while the peer state is still `PeerState::Applying`
### What version of TiKV are you using?

Matser

### What operating system and CPU are you using?

Not related

### Steps to reproduce

A peer with the peer state set to `PeerState::Applying` is started the apply snapshot process. If before it actually starts to apply snapshot data, it received a message (gc message or raft msg with larger peer id) telling it to destroy itself, the peer will try to cancel applying snapshot and submit the change of peer state (to `PeerState::Tombstone`) to the `WriteBatch`.

At the meantime, a canceled snapshot will be gc by the `StoreTick::SnapGc` tick, which will delete the snapshot files immediately and the `WriteBatch` may not persist yet. If TiKV then crashes and recover, the peer state is still `PeerState::Applying` and TiKV will panic because the snapshot does not exist.

This case can happen in scenarios like a TiKV is down while there are regions applying snapshot, and recover after a long time, by then these regions will try to continue to apply snapshot and the leader will tell them to gc themself, at this time the TiKV is crashed.

### What did you expect?

Not panic

### What did happened?

Panic because snapshot files not exist

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #11782 to release-5.2
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/11907
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/11907:release-5.2-8c31f1a392ba
```

---

skip deleting snapshot files in peer pending_remove is true and the snapshot is being applied and canceled -- close #11746

This is to avoid the potential panic when the snapshot files are deleted, but the peer's status (Tombstone) is not persisted in disk due to tikv crash.

Signed-off-by: tonyxuqqi <tonyxuqi@outlook.com>

### What is changed and how it works?

Close #11746 

What's Changed:
Skip deleting snapshot files in peer.rs when the snapshot is being applied and just canceled and the peer is pending removal. 
### Check List 

Tests 

- Integration test

Side effects

### Release note 

```release-note
Fix a potential panic (#11746) when snapshot files have been deleted but the peer's status is still Applying.
```

### CODE REVIEW

File: components/raftstore/src/store/fsm/peer.rs [hunk 1]
```diff
@@ -2072,6 +2075,35 @@ where
 
     fn destroy_peer(&mut self, merged_by_target: bool) {
         fail_point!("destroy_peer");
+<<<<<<< HEAD
+=======
```
Comment (MyonKeminta) [inline components/raftstore/src/store/fsm/peer.rs:2079]: Please resolve conflicts

Comment (tonyxuqqi) [reply components/raftstore/src/store/fsm/peer.rs]: > Please resolve conflicts

done.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by BusyJay — APPROVED** [review]

**Pull request merged**